app/controllers/activities_controller.rb

```diff
@@ -20,6 +20,14 @@ def show
     @activity_history = ActivityHistory.new
   end
 
+  def destroy
+    @activity = Activity.find(params[:id])
+    @activity.destroy_all
```

Comment:
```suggestion
    @activity.destroy
```

models/activity の `has_many :activity_histories` に `dependent: :destroy` を足せばいけるはず！

Reply:
ありがとう〜